src/librustc_mir/transform/const_prop.rs

```diff
@@ -413,7 +410,7 @@ impl<'mir, 'tcx> ConstPropagator<'mir, 'tcx> {
                         diagnostic.report_as_lint(
                             self.ecx.tcx,
                             "this expression will panic at runtime",
-                            lint_root,
+                            self.lint_root?,
```

Comment:
Please use the lint_root function and remove the field

Reply:
Done and rebased again on top of master.